js/PivotReport.PivotTable.js

```diff
@@ -276,30 +292,36 @@ CRM.PivotReport.PivotTable = (function ($) {
       }]
     };
 
-    CRM.api3(apiCalls).done(function (result) {
-      that.dateFields = result.dateFields.values;
-      that.relativeFilters = result.relativeFilters.values;
-      that.header = result.getHeader.values;
-      that.total = parseInt(result.getCount.values, 10);
-      that.pivotCount = parseInt(result.getPivotCount.values, 10);
-      that.crmConfig = result.getConfig.values[0];
-
-      $.each(that.dateFields, function (i, value) {
-        that.config.derivedAttributes[value + ' (' + ts('per month') + ')'] = $.pivotUtilities.derivers.dateFormat(value, '%y-%m');
-      });
+    $.when(
```

Comment:
Is it necessary to resolve `this.resolveCustomFilterDefaultValues()`before calling the done callback? I mean is it dependant?